### PR TITLE
Added Iterative Quick Sort 

### PR DESCRIPTION
**Pull request by sebe324**
#### Description of Change
I've added an implementation of iterative quick sort.
It uses the stack instead of recursion.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by sebe324** [comment]
can someone check this pr out? @realstealthninja

File: sorting/quick_sort_iterative.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/**
+ * @file
+ * @brief Quick Sort without recursion. This method uses the stack instead.
+ * Both recursive and iterative implementations have O(n log n) best case
+ * and O(n^2) worst case.
+ * @details
+ * https://stackoverflow.com/questions/12553238/quicksort-iterative-or-recursive
+ * https://en.wikipedia.org/wiki/Quicksort
+ * https://www.geeksforgeeks.org/iterative-quick-sort/
+ * @author Sebe324 (https://github.com/sebe324)
```
Comment (realstealthninja) [inline sorting/quick_sort_iterative.cpp:10]:
```suggestion
 * @author [Sebe324](https://github.com/sebe324)
```
For hyperlinking purposes

File: sorting/quick_sort_iterative.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ */
+
+#include <iostream> //for std::cout
+#include <vector> //for std::vector
+#include <stack> //for std::stack
+#include <algorithm> //for std::is_sorted
+#include <cassert> //for assert
```
Comment (realstealthninja) [inline sorting/quick_sort_iterative.cpp:17]:
```suggestion
#include <iostream> /// for std::cout
#include <vector> /// for std::vector
#include <stack> /// for std::stack
#include <algorithm> ///for std::is_sorted
#include <cassert> /// for assert
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]
Clean documentation. Well written too! Thank you for this contribution ♥️

File: sorting/quick_sort_iterative.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+    tests();
```
Comment (realstealthninja) [inline sorting/quick_sort_iterative.cpp:123]:
```suggestion
    tests(); // run self test implementation
```

**Review by realstealthninja — APPROVED** [review]

**Review by realstealthninja — DISMISSED** [review]

File: sorting/quick_sort_iterative.cpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+ * https://stackoverflow.com/questions/12553238/quicksort-iterative-or-recursive
+ * https://en.wikipedia.org/wiki/Quicksort
+ * https://www.geeksforgeeks.org/iterative-quick-sort/
+ * @author [Sebe324](https://github.com/sebe324)
+ */
+
+#include <iostream> /// for std::cout
+#include <vector> /// for std::vector
+#include <stack> /// for std::stack
+#include <algorithm> ///for std::is_sorted
```
Comment (realstealthninja) [inline sorting/quick_sort_iterative.cpp:16]:
```suggestion
#include <algorithm> /// for std::is_sorted
```

File: sorting/quick_sort_iterative.cpp [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+ * https://www.geeksforgeeks.org/iterative-quick-sort/
+ * @author [Sebe324](https://github.com/sebe324)
+ */
+
+#include <iostream> /// for std::cout
+#include <vector> /// for std::vector
+#include <stack> /// for std::stack
+#include <algorithm> /// for std::is_sorted
+#include <cassert> /// for assert
+
```
Comment (realstealthninja) [inline sorting/quick_sort_iterative.cpp:18]:
```suggestion


/**
 * @namespace sorting
 * @brief Sorting algorithms
 */
namespace sorting {
```
should be inside sorting namespace

File: sorting/quick_sort_iterative.cpp [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+        if(pivotIndex+1<end)
+        {
+            stack.push(pivotIndex+1);
+            stack.push(end);
+        }
+    }
+}
+
```
Comment (realstealthninja) [inline sorting/quick_sort_iterative.cpp:82]:
```suggestion

} // namespace sorting
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Comment by realstealthninja** [comment]
Tests failing! The issue is the functions youre using inside test is not the namespaced one

Fix is prefixing the function calls with
```cpp
   sorting::function_name()
```

**Comment by sebe324** [comment]
Oh I forgot, I'll fix it later today.

**Review by realstealthninja — APPROVED** [review]
Looks good

**Pull request merged**